File: sdk/python/packages/flet/src/flet/controls/reorderable_draggable.py
from flet.controls.adaptive_control import AdaptiveControl
from flet.controls.constrained_control import ConstrainedControl
from flet.controls.control import Control, control


@control("ReorderableDraggable")
class ReorderableDraggable(ConstrainedControl, AdaptiveControl):
    index: int
    content: Control

    def before_update(self):
        super().before_update()
        assert self.content.visible, "content must be visible"

File: packages/flet/lib/src/controls/reorderable_draggable.dart
import 'package:flutter/material.dart';

import '../extensions/control.dart';
import '../models/control.dart';
import '../widgets/error.dart';
import '../widgets/reorderable_item_scope.dart';

class ReorderableDraggableControl extends StatelessWidget {
  final Control control;

  const ReorderableDraggableControl({super.key, required this.control});

  @override
  Widget build(BuildContext context) {
    debugPrint("ReorderableDraggableControl build: ${control.id}");

    var scope = ReorderableItemScope.of(context);
    var index = scope?.index;
    if (index == null) {
      return const ErrorControl(
          "ReorderableDraggable must be placed inside ReorderableListView.");
    }
    var content = control.buildWidget("content");
    if (content == null) {
      return const ErrorControl(
          "ReorderableDraggable.content must be set and visible");
    }

    return ReorderableDragStartListener(
        index: index, enabled: !control.disabled, child: content);
  }
}
